### PR TITLE
Adding release checklist

### PR DESCRIPTION
**Pull request by jlk**
Initial PR for a release checklist.

Creating PR now to get feedback - let me know any thoughts, or feel free to contribute.

I know this is missing integration testing - if somebody's bored, feel free to add, otherwise I'll get that in soon.

### CODE REVIEW

File: release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Release Checklist
+This file provides a guideline on steps that need to be taken to properly release a new version of Terrascan:
+
+### Run unit & integration tests
+The following command will kick off both unit tests and end-to-end tests. These tests should finish clean before a release is made:
+
+```
+make test
```
Comment (dev-gaur) [inline release_checklist.md:8]:
```suggestion
make cicd
```

Comment (dev-gaur) [reply release_checklist.md:8]: this will also test the docker build.

Comment (dev-gaur) [reply release_checklist.md:8]: And in future we can add subtargets like `acceptance tests` and `e2e-tests` under the make `cicd` target

File: release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### Tag the new release with git
+Once the changelog PR has been merged, pull the updated code, tag it with the new version number, and push the tag back to the repo:
+
+```
+git pull
+git tag v1.5.0
+git push
+```
+(again, subsitute v1.5.0 for the appropriate version being released)
```
Comment (dev-gaur) [inline release_checklist.md:35]: Maybe put this NOTICE above the code block ? 😅 or use a place holder like <NEW_TERRASCAN_VERSION> instead of v1.5.0 .

**Review by dev-gaur — CHANGES_REQUESTED** [review]
minor changes requested. thanks for the initiative.

**Review by dev-gaur — APPROVED** [review]

**Comment by kanchwala-yusuf** [comment]
Thank you @jlk for this PR!

**Pull request merged**